coral-component-colorpicker/src/scripts/ColorArea.js

```diff
@@ -0,0 +1,560 @@
+ @htmltag coral-colorpicker-colorarea
+ @extends {HTMLElement}
+ @extends {BaseComponent}
+ */
+class ColorArea extends BaseComponent(HTMLElement) {
+  constructor() {
+    super();
+
+    this._delegateEvents(commons.extend(this._events, {
+      'key:up': '_handleKey',
+      'key:right': '_handleKey',
+      'key:down': '_handleKey',
+      'key:left': '_handleKey',
+      'key:pageUp': '_handleKey',
+      'key:pageDown': '_handleKey',
+      'key:home': '_handleKey',
+      'key:end': '_handleKey',
+      
+      'input': '_onInputChangeHandler',
+      
+      'touchstart': '_onMouseDown',
+      'mousedown': '_onMouseDown',
+      'capture:focus': '_focus',
+      'capture:blur': '_blur'
+    }));
+    
+    // Templates
+    this._elements = {};
+    colorArea.call(this._elements, {commons, i18n});
+    // default values
+    this._x = 1;
+    this._y = 1;
+    this._hue = 120;
+  }
+
+  /** @ignore */  
+  render() {
+    super.render();
+
+    this.classList.add(CLASSNAME);
+
+    const frag = document.createDocumentFragment();
+
+    // Render template
+    frag.appendChild(this._elements.colorAreaGradient);
+    frag.appendChild(this._elements.colorHandle);
+    frag.appendChild(this._elements.sliderX);
+    frag.appendChild(this._elements.sliderY);
+
+    // Support cloneNode
+    while (this.firstChild) {
+      const child = this.firstChild;
+      if (child.nodeType === Node.ELEMENT_NODE && child.hasAttribute('handle')) {
+        this.removeChild(child);
+      } 
+      else {
+        frag.appendChild(child);
+      }
+    }
+
+    this.appendChild(frag);
+
+    // These should be used to set a property since property handler aren't called until elements are attached to dom.
+    // Attribute values are delivered to change-listeners even if element isn't attached to dom yet, so attributes 
+    // can be set to e.g. this._elements.colorHandle.
+    this._handle = this.querySelector('._coral-ColorPicker-ColorArea-colorHandle');
+    this._sliderX = this.querySelector('._coral-ColorPicker-ColorArea-slider[name="x"]');
+    this._sliderY = this.querySelector('._coral-ColorPicker-ColorArea-slider[name="y"]');
+    this._gradient = this.querySelector('._coral-ColorPicker-ColorArea-gradient');
+    this._updateHue(this._hue);
+    this.x = this._x;
+    this.y = this._y;
+    this._updateHandle(this._hue, this.x, this.y, this.color);
+    this._reflectAttribute('color', this.color);
+  }
+
+  /**   
+   The ColorArea label.
+   @default ''
+   @type {String}
+   @htmlattribute label
+   @htmlattributereflected
+   */   
+  get label() {
+     return this._label;
+  }
+  
+  set label(value) {
+    this._label = value;
+    this._reflectAttribute('label', this._label);
+    this._syncInputSliderAttrs();
+  } 
+
+  /**   
+   The ColorArea x value. value should be in multiple of x-step size.
+   @default 1
+   @type {String}
+   @htmlattribute label
+   @htmlattributereflected
+   */   
+  get x() {
+    return this._x;
```

Comment:
in case _x is undefined we need to return default value.

Reply:
It is initialised in constructor.